lua/rustaceanvim/rust_analyzer.lua

```diff
@@ -5,18 +5,46 @@ local os = require('rustaceanvim.os')
 ---@class rustaceanvim.rust-analyzer.ClientAdapter
 local M = {}
 
+M.load_os_rustc_target = function()
+  vim.system({ 'rustc', '-Vv' }, { text = true }, function(result)
+    if result.code == 0 then
+      for line in result.stdout:gmatch('[^\r\n]+') do
+        local host = line:match('^host:%s*(.+)$')
+        if host then
+          M.os_rustc_target = host
+          break
+        end
+      end
+    end
+  end)
+end
+
+---@class rustaceanvim.lsp.get_clients.Filter: vim.lsp.get_clients.Filter
+---@field exclude_rustc_target? string Cargo target triple (e.g., 'x86_64-unknown-linux-gnu') to filter rust-analyzer clients
+
 ---@param bufnr number | nil 0 for the current buffer, `nil` for no buffer filter
----@param filter? vim.lsp.get_clients.Filter
+---@param filter? rustaceanvim.lsp.get_clients.Filter
 ---@return vim.lsp.Client[]
 M.get_active_rustaceanvim_clients = function(bufnr, filter)
   ---@type vim.lsp.get_clients.Filter
-  filter = vim.tbl_deep_extend('force', filter or {}, {
+  local client_filter = vim.tbl_deep_extend('force', filter or {}, {
     name = 'rust-analyzer',
   })
   if bufnr then
-    filter.bufnr = bufnr
+    client_filter.bufnr = bufnr
+  end
+  local clients = vim.lsp.get_clients(client_filter)
+  if filter and filter.exclude_rustc_target then
+    clients = vim.tbl_filter(function(client)
+      local cargo_target = vim.tbl_get(client, 'config', 'settings', 'rust-analyzer', 'cargo', 'target')
+      if filter.exclude_rustc_target == DEFAULT_RUSTC_TARGET and cargo_target == nil then
```

Comment:
issue: `DEFAULT_RUSTC_TARGET` is in the `rustc` module (ditto for the other usages).

tip: If you have Nix installed, I recommend you enter a nix devShell (`nix develop`) and run `pre-commit run --all-files`.
It will run all the type checks and lints needed to catch these issues.
Otherwise, you can run `luacheck` and the `lua-language-server` CLI to get the lints without using Nix.

Reply:
thanks!